source/docs/software/hardware-apis/sensors/gyros-software.rst

```diff
@@ -157,6 +191,16 @@ Displaying the robot heading on the dashboard
             frc::Shuffleboard.GetTab("Example tab").Add(gyro);
         }
 
+    .. code-block:: python
+
+        from wpilib.shuffleboard import Shuffleboard
+
+        # Use gyro declaration from above here
+
+        def robotInit(self):
+            # Places a compass indicator for the gyro heading on the dashboard
+            Shuffleboard.getTab("Example tab").add(gyro)
```

Comment:
This isn't correct. If you're going to use `gyro`, it needs to be declared in robotInit, it can't just stand by itself. Additionally, you need to store the gyro in self, there's no equivalent to declaring variables at class scope in RobotPy.

See the difference between these two examples:

* https://github.com/robotpy/examples/blob/411d9b02dfa1c4ec5d35410a40f88be796f82f70/gyro/robot.py#L39
* https://github.com/wpilibsuite/allwpilib/blob/d9b504bc84019da6b2da3293d13264f2b01d2c4c/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/gyro/Robot.java#L34

Everything at class scope in Java *must* be in some function in python. We recommend teams do it in `robotInit`.

Reply:
I don't understand what you want here.  If it is self.gyro don't you think it is fair to assume that it has been created above?